### ISSUE TITLE
M> RChain Life - a learning game

### ISSUE THREAD

**Issue by dckc** [rchain/bounties#259]
somewhat like [The Game of Life](https://en.wikipedia.org/wiki/The_Game_of_Life) mixed with role playing games. You roll dice and move around the board where you learn lessons about cryptocurrency, distributed systems, etc.

brainstorm:

- [RChain Life](https://docs.google.com/presentation/d/1dW1nsNDJjXBd6YTBOxQ21GBGqHMt5Q8f3B1RIG3bNao/edit#slide=id.p) slides.

Background: a friend of mine said his child, around age 12, asked him about mining bitcoin. And Jan 12, @patrick727 proposed a learning game for the governance forum https://github.com/rchain/Members/issues/189#issuecomment-357315780 . And principle 5 of coops is Education, Training And Information.

**Comment by zsluedem** [comment]
interesting~

**Comment by Ojimadu** [comment]
Interesting idea. I did not understand how "cryptography 101" worked on the game. May be I don't understand how modular arithmetic works.
Does the game end at cryptokitties? How about adding "forks" to it?

**Comment by dckc** [comment]
@Ojimadu I changed the presentation of modular arithmetic to a table. Does it make more sense now?

Are you familiar with https://en.wikipedia.org/wiki/Remainder ?

And no, the game is not intended to end at cryptokiddies. I added a "TODO: envision RChain in action" at the end to clarify.

**Comment by dckc** [comment]
@Keaycee why add the `M>` prefix? Is that prefix consumed by some automated mechanism? Surely for humans, the label suffices.

@Ojimadu I am somehow uncomfortable calling this marketing. I'm inclined to make an Education label to hark to number 5. **Education, Training and Information** in the [principles of coops](https://ica.coop/en/whats-co-op/co-operative-identity-values-principles).

**Comment by BelovedAquila** [comment]
@dckc I just created those labels (education, training and information), how about that?

**Comment by dckc** [comment]
I figured out what the objective of the game should be: **participate in RChain applications at many scales** such as family, city, marketplace, gaming community, sports fan club, charitable organizations, nation, ...

See also TimBL's [The Scale-free nature of the Web](https://www.w3.org/DesignIssues/Fractal).

**Comment by dckc** [comment]
@pmoorman would you please look at this game and see what you think of it as a vehicle for communicating RChain values and some of the technical stuff?

I'm not sure if it's going to literally turn into a game or just remain a slide presentation.

I hope to use it as a lightning talk at the gov forum.

**Comment by Ojimadu** [comment]
@dckc The presentation change helped. I now understand the modular arithmetic aspect. Thanks.

**Comment by pmoorman** [comment]
@dckc I think the idea of a game is great. The main concern I have is: _"is this the best way to spend our time?"_ And then the answer would probably be: _"No, there are much bigger marketing issues to solve first."_ It doesn't fit into the 80/20-rule, if you know what I mean.

However...    (plot twist!)

I still see a great use-cases for this game. The thing I like best about this draft is that it forces people to think about the _basics_ behind a blockchain. 

Your _chronological narrative_ with a focus on the _basics_, creates a CONTEXT that could be extremely fertile soil for our marketing. If we do it well, the people that go through it will forget about "practical details" for a second, and ask themselves **fundamental** questions, like... _"does this whole blocks-in-a-chain approach even make sense?"_

I think it's great if we can turn this into an interactive sort of presentation, where **the basic story arch could be:**

1. Here is how crypto got started. This is "crypto 101".

2. It's a little game. Definitely not a _sales pitch_ (God forbid!). Let's all play along for a second... 

3. {{ insert some basic chronological game elements here. Like you did in the slide deck }}

4. As the game progresses, we run into a problem...  (scalability)

5. **["aha moment #1"]** Maybe... this entire approach of everything-in-sequence, is kinda weird.
 Imagine that the internet would be sequential, and that _every web page_ could only be served _sequentially_ after all previous requests had been dealt with. =>  **No wonder blockchains don't scale...** 

6. The only _obvious_ approach — something a smart 8th grader could come up with — is that transactions need to somehow start happening in _parallel_, or concurrently.

7. Turns out that most programming languages really struggle with this (oh oh, problem!)

8. **[Aha moment #2]:** But... in a small backcorner of the world, a solution exists  =>  Rholang / functional programming / Rho-calculus to the rescue

9. {{ little game elements to appreciate how concurrency works in Rholang (??) }}

10. Conclusion: RChain is a blockchain built — from the ground up — to support this concurrency-oriented approach. RChain is the only logical next step forward from here (*finishes the chronological storyline). 

--------

Oke, so maybe I went a little off-topic here, but when I read your slide deck, it really got me thinking how we could fit the story of RChain into a chrological narrative that absolute beginners could understand.

The game elements are great, because they:

... Establish the presenter as an in-the-moment authority for recapping the current situation so nicely...

... Let people let their guards down, and consider _fundamental_ alternative solutions for a second...

... And let's people _engage with the story_, in a way that makes them feel like _they_ came up with this idea all along.

+++++

That's my 2 cents  ;-)
Hope it helps!

**TL;DR:** I wouldn't turn it into a board game, but the journey you're presenting is a great template for super-engaging live presentations (and potentially other marketing formats).

**Comment by dckc** [comment]
@pmoorman I'm interested to know what you think of the delivery of the talk; it was in this session:

 - [Rchain Governance Conference Day 2 Session 1](https://www.youtube.com/watch?v=Mmkae9E93tk&feature=youtu.be&t=6393)

I had fun giving the talk, but I don't know about watching a recording... watching myself on video is rarely fun.

**Comment by dckc** [comment]
I don't see myself making more time for this any time soon.
